database/migrations/default/1728066705364_drop_crashes_edits_update_crashes_cris_functions/up.sql

```diff
@@ -0,0 +1,205 @@
+        new.is_deleted,
+        new.is_temp_record,
+        new.latitude,
+        new.light_cond_id,
+        new.longitude,
+        new.medical_advisory_fl,
+        new.obj_struck_id,
+        new.onsys_fl,
+        new.private_dr_fl,
+        new.road_constr_zone_fl,
+        new.road_constr_zone_wrkr_fl,
+        new.rpt_block_num,
+        new.rpt_city_id,
+        new.rpt_cris_cnty_id,
+        new.rpt_hwy_num,
+        new.rpt_hwy_sfx,
+        new.rpt_rdwy_sys_id,
+        new.rpt_ref_mark_dir,
+        new.rpt_ref_mark_dist_uom,
+        new.rpt_ref_mark_nbr,
+        new.rpt_ref_mark_offset_amt,
+        new.rpt_road_part_id,
+        new.rpt_sec_block_num,
+        new.rpt_sec_hwy_num,
+        new.rpt_sec_hwy_sfx,
+        new.rpt_sec_rdwy_sys_id,
+        new.rpt_sec_road_part_id,
+        new.rpt_sec_street_desc,
+        new.rpt_sec_street_name,
+        new.rpt_sec_street_pfx,
+        new.rpt_sec_street_sfx,
+        new.rpt_street_desc,
+        new.rpt_street_name,
+        new.rpt_street_pfx,
+        new.rpt_street_sfx,
+        new.rr_relat_fl,
+        new.schl_bus_fl,
+        new.surf_cond_id,
+        new.surf_type_id,
+        new.thousand_damage_fl,
+        new.toll_road_fl,
+        new.traffic_cntl_id,
+        new.txdot_rptable_fl,
+        new.updated_by,
+        new.wthr_cond_id
+    );
+
+    RETURN NULL;
+END;
+$$;
+
+create or replace trigger insert_new_crashes_cris
+after insert on public.crashes_cris
+for each row
+execute procedure public.crashes_cris_insert_rows();
+
+-- recreating the crashes_cris_update function to use just the crashes unified layer
+-- to check if VZ has edited a value and remove the crashes_edits part
+drop trigger if exists update_crashes_from_crashes_cris_update on public.crashes_cris;
+
+drop function if exists public.crashes_cris_update;
+
+--
+-- handle a cris crashes update by updating the
+-- unified crashes record from cris + edit values
+--
+create or replace function public.crashes_cris_update()
+returns trigger
+language plpgsql
+as $$
+declare
+    new_cris_jb jsonb := to_jsonb (new);
+    old_cris_jb jsonb := to_jsonb (old);
+    unified_record_jb jsonb;
+    column_name text;
+    updates_todo text [] := '{}';
+    update_stmt text := 'update public.crashes set ';
+begin
+    -- get corresponding the VZ record as jsonb
+    SELECT to_jsonb(crashes) INTO unified_record_jb from public.crashes where public.crashes.id = new.id;
+
+    -- for every key in the cris json object
+    for column_name in select jsonb_object_keys(new_cris_jb) loop
+        -- ignore audit fields
+        continue when column_name in ('created_at', 'updated_at', 'created_by', 'updated_by');
+        -- if the new cris value doesn't match the old cris value
+        if(new_cris_jb -> column_name <> old_cris_jb -> column_name) then
+            -- see if the unified record has the same value as the old cris value
+            if (unified_record_jb -> column_name = old_cris_jb -> column_name) then
```

Comment:
this line is updated. now we are checking in the unified record value is equal to the old cris value, if so then there hasnt been a user edit so we want update the unified record with the new cris value.

Reply:
cool! this makes sense to me, and it is pretty nice to take advantage of the availability of the new and old values of the CRIS record and compare to the current `crashes` record to determine what is or is not an edit. 😎 

It's been a while since I've looked at this code so it was helpful for me to see the change isolated to the crashes table too. 💯